src/spec/expectations.cr

```diff
@@ -286,41 +277,40 @@ module Spec
     # If *message* is a regular expression, it is used to match the error message.
     #
     # It returns the rescued exception.
-    macro expect_raises(klass, message, file = __FILE__, line = __LINE__)
-      %failed = false
+    def expect_raises(klass : T.class, message = nil, file = __FILE__, line = __LINE__) forall T
+      failed = false
       begin
-        {{yield}}
-        %failed = true
-        fail "Expected {{klass.id}} but nothing was raised", {{file}}, {{line}}
-      rescue %ex : {{klass.id}}
+        yield
+        failed = true
```

Comment:
Can't we skip this `failed` hackery and implement it with an `else` clause in the `begin` block? Then we could inline the `begin`/`end` block into the method.

Reply:
but what about:

```
        fail "Expected #{klass} but nothing was raised", file, line
```

we want to ignore that exception

Reply:
Thanks, done!